docs/orchestration/recipes/third_party_auth.md

```diff
@@ -0,0 +1,86 @@
+
+```python
+from prefect import Flow
+from prefect.environments.storage import GCS
+
+
+storage = GCS(bucket="my-bucket", 
+              secrets=["GCP_CREDENTIALS"])
+
+f = Flow("empty-flow", storage=storage)
+```
+
+Once this flow is registered and run through an Agent, the first thing it will do is pull the value of the `"GCP_CREDENTIALS"` secret from Prefect Cloud and place it into `prefect.context.secrets` under the appropriate key.  These credentials are now available for any task or Prefect API call to Google, _including for pulling the Flow itself from GCS_.
+
+Any number of secrets can be declared on your `Storage` option.
+
+::: warning This only applies to Prefect built-ins
+This off-the-shelf experience generally only applies to interfaces that are natively included as a part of the Prefect package. Note that if you write a completely custom task for interacting with a third party service, you will also need to consider how this task will authenticate.  Whether you choose to use Prefect Secrets or another option is up to you.
+:::
+
+#### Passing Secrets from an Agent
+
+Recall that users can autopopulate Prefect context with values through the use of environment variables.  In particular, all secrets can be set by providing an appropriate value to `PREFECT__CONTEXT__SECRETS__XXXX` in your runtime environment.  One way of achieving this through Prefect is by configuring your agent(s) to pass the appropriate value to each flow run it submits.  Using a Docker Agent as an example:
+
+```
+prefect agent start docker \\
+    -e PREFECT__CONTEXT__SECRETS__AWS_CREDENTIALS=${AWS_CREDENTIALS}
+```
+
+This will then ensure that the `AWS_CREDENTIALS` secret is globally present in `prefect.context` for all flow runs submitted through this agent; in this case, it's value will be whatever value the `AWS_CREDENTIALS` environment variable has in the agent's own environment (which itself might have been set through some secure mechanism such as a Kubernetes Secret).
+
+### Providing Secrets on a per-task basis
+
+In some instances you may need to override the global options presented above on a per-task basis, or you may choose to explicitly provide credentials to every task that needs them.  Either way, all Prefect tasks from the Task library which require credentials to authenticate offer optional runtime arguments for providing this information.
+
+
+### List of default Secret names
+
+The following is a list of the default names and contents of Prefect Secrets that, if set and declared, can be used to automatically authenticate your flow with the listed service:
+
+- `GCP_CREDENTIALS`: a dictionary containing a valid [Service Account Key](https://cloud.google.com/docs/authentication/getting-started)
+- `AWS_CREDENTIALS`: a dictionary containing two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY` which are passed directly to the `boto3` client
```

Comment:
We should link to https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html and also expand the default `AWS_CREDENTIALS` to accept a session token

Reply:
@joshmeek updated per all of your requests